### DISCUSSION TITLE
Home Assistan und Hichi, Daten-Spikes in HA

### DISCUSSION THREAD

**Discussion by DerWer32** [hoylabs/OpenDTU-OnBattery#851] [Q&A]
Hallo,
ich nutze seit kurzem OpenDTU-OnBattery statt OpenDTU um die Nulleinspesung zu nutzen.
Als Strommeter habe den Hichi.
Beides ist auch in HA eingebunden. 
Hichi mit der Tasmota-Integration an HA und wie beschrieben via HTTP/JSON an OpenDTU-OnBattery.
OpenDTU-OnBattery mit MQTT-Integration an HA angebunden.
Funktioniert soweit auch alles wie genünscht, nur ab und an habe ich vom Hichi Daten-Spikes.
Diese Spickes (flasche Werte z.B. des Stromzählers) zerschießen mir die Anzeige im Energie-Daschbord.
Dies ist sehr ärgerlich da ich dann keine Übersicht mehr über die Tage habe.
Mit OpenDTU hatte ich diese falschen Werte nicht bis jetzt.
Ich vermute das die gleichzeitige Abfrage von OpenDTU-OnBattery und HA sich irgendwann ins Gehege kommen.

Hat jemand vielleicht auch schon dieses Prolem gehabt, bzw. eine Lösung?

Ich habe schon an den Zeiten für die Übertragung zu HA im Hichi gespielt aber ohne Erfolg.
SG.
![Unbenannt](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/120647356/a2d74c31-88eb-43a3-8d72-8d210b2650db)

**Comment by DerWer32** [comment]
Hier noch ein das passende Bild des Verlaufs des Stromzählers in HA.

![Unbenannt2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/120647356/e23c73cd-21c0-4c05-b69f-d31fcbade3ee)

**Comment by spcqike** [comment]
Ich vermute du hast fehlauslesungen. Sitzt der Lesekopf sauber und sicher? 

Ich hatte sowas mit meinen tasmota auch. Lag definitiv an der Positionierung des Empfängers. 

Tasmota selber würd ich als Ursache ausschließen. Da geht weit mehr als 2 Abfragen aller 10 Sekunden.

**Comment by DerWer32** [comment]
Ok, danke.
Ich werde den Lesekopf mal kontrllieren und ausrichten.

**Comment by ottelo9** [comment]
@DerWer32 
Du hast den Hichi Sensor mit Tasmota oder? Dann kannst du doch im Tasmota Script den Filter aktivieren. Das sollte helfen:

Das habe ich hier auf meinem Blog beschrieben:
https://ottelo.jimdofree.com/stromz%C3%A4hler-auslesen-tasmota/#4d

+1,3,s,16,9600,MT175
16 = Werte werden Median gefiltert = geglättet.

**Comment by DerWer32** [comment]
Hallo,
leider hat die Positionierung keinen Erfolg gebracht, oder ich hatte nicht das richtige Händchen.
Von den beiden Löchern im Hichi und den Dioden im Zähler her sollte es passen. Um sichtbare Aussetzer zu sehen habe ich beim bewegen doch recht viel Spielraum.
Denke aber die Position sollte passen.

@ottelo9 : Danke für den Link. Werde nach deiner Beschreibung mal testen.
Wird dadurch dann auch der momentane Leistungswert gemittelt der für die Dynamic Power Limiter -Funktion verwendet wird? 
Dann würde sich auch die Reaktionszeit erhöhen wenn das Limit des WR höher gesetzt werden soll?

![Unbenannt](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/120647356/0b0e9736-43be-4364-b0aa-6fa3793e0626)
![Unbenannt2](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/120647356/6f73c3d3-69c1-442d-b19d-e4ac7b15c57b)

**Reply by ottelo9** [reply]
Ja eine leichte Verzögerung wirst du dann bekommen. Aber das ist kein Problem.

**Comment by Manos1966** [comment]
@DerWer32 welche update Geschwindigkeit hast du eingestellt? Real Time?

In den Tasmota Loggin Einstellungen kannst du die Teleperiod Zeit zwischen 10-300 Sekunden stellen.
Als Standart steht 300 drin. Daher sendet Tasmota alle 5 Minuten wenn man es nicht ändert
Wenn du häufiger als 10 Sekunden senden möchtest musst du die **descriptor** ändern, **dann wird bei jeder Werte Änderung gesendet**
1,77070100100700ff@1,P,W,P,**18**  
the last entry is the number of digits + an optional flag to send the value as fast as possible code = 16 so 16+2 = 18 means 2 digits + fast send

Beispiel:
Wenn deine Zeile so aussieht:  1,77070100100700ff@1,P,W,P,**0** 
rechnest du 0+16=16 und änderst es so: 1,77070100100700ff@1,P,W,P,**16**
Also immer die letzte Zahlt in Descriptor plus oder minus 16.

**Comment by DerWer32** [comment]
Das Tasmota Script habe ich angepasst:
+1,3,o,**0**,9600,P -> +1,3,o,**16**,9600,P
Von gestern bis jetzt keine Sprünge mehr.

Die Teleperiod Zeit steht auf 10 Sekunden.
Der Discriptor ist für alle Werte normal eingetragen.
Für die Leistung z.B. 1,1-0:16.7.0*255(@1,Leistung,W,Cur_Power,**2**

In der Console  von Tasmota des Stromzähler sieht man folgende Einträge:
![Unbenannt](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/120647356/58fc16fc-731d-4459-99bf-4f1a2c2763e3)

Die Einträge "STATE" und "SENSOR" gehen mit der Teleperiod Zeit mit MQTT an den HA denke ich.
Der Eintrag "STATUS8" kommt also vermutlich von OpenDTUonBattery.
Abend, wenn die WR aus sind, habe ich beobachtet das "STATUS8" auch nur alle ca. 10sek. erscheint.

**Reply by Manos1966** [reply]
> Das Tasmota Script habe ich angepasst:
+1,3,o,0,9600,P -> +1,3,o,16,9600,P
Von gestern bis jetzt keine Sprünge mehr.

Also, Problem geloescht? 😃

**Comment by DerWer32** [comment]
Für das Problem gibt es erstmal Entwarnung.
Beim Stromzähler habe ich keine Ausreißer mehr gesehen.
Das Energie-Dashbord und die Energieverteilung ist wieder brauchbar.

Zwei Sachen habe ich noch beobachtet:
- Ich sehe noch Ausreißer bei der momentanen Leitung. Was mich aber nicht weiter stört.
![Unbenannt](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/120647356/91039dfa-1435-40ea-9e8e-e98a387bc95d)

- In der zwischenzeit hat ein Kollege auch OpenDTU auf OpenDTUonBattery mit Anbindung eines Hichi umgestellt.
  Bei ihm sind nach der Umstellung auch die Spitzen aufgetreten die vorher mit OpenDTU definitiv nicht da waren. 
  Abhilfe war auch hier die 0 auf die 16 umzustellen beim Tasmota.

Danke für die Hilfe. SG.

**Reply by ottelo9** [reply]
Das hört sich ja gut an. Aber die Ausreißer (Leistung) sind wirklich nicht echt? Ich hab zb bei meinem E-Auto wenn es noch dran steckt aber voll ist ab und zu solche Peaks.

**Reply by Manos1966** [reply]
Danke für die Info @DerWer32 ,  ich werde es im Wiki notieren 👍

**Comment by gone-for-coding** [comment]
Ich hatte exakt dasselbe Problem und konnte es mit einer CRC-Überprüfung samt größerem Buffer lösen. Darauf hat mich ein Entwickler im Tasmota-Projekt hingewiesen. Dort habe ich auch den Hinweis bekommen, wie man nur einzelne Werte per Rest abfragen kann.

Siehe hier: https://github.com/arendst/Tasmota/discussions/23342

**Reply by Manos1966** [reply]
Danke @gone-for-coding , addiert https://github.com/hoylabs/OpenDTU-OnBattery/wiki/Power-Meter#important-for-tasmota-users
